operations.go

```diff
@@ -112,6 +124,9 @@ func newOpsType(stack *stackType) *opsType {
 
 		ophandler{"fac", "Calculate factorial of x", 1, func(a []float64) ([]float64, int, error) {
 			x := uint64(a[0])
+			if float64(x) != a[0] {
```

Comment:
Instead of doing this maneuver with uint64 and float to make sure the number is an integer, maybe use `if math.Floor(a) != a { ... }` like elsewhere in the code. This avoids the confusing type casting.